constraints/recursiveCatEvals.js

```diff
@@ -56,26 +67,53 @@ var sCat = ["cp", "xp", "x0"];
  * 7/29/19 refactor of an earlier version
  */
 
-function markMinMax(mytree){
+function markMinMax(mytree, options){
+	if(options.requireLexical){
+		mytree = createDummies(mytree, 'func');
+	}
+	if(options.requireOvertHead){
+		mytree = createDummies(mytree, 'silent');
+	}
+	return markMinMaxInner(mytree, options)
+}
+
+function markMinMaxInner(mytree, options){
 	/* If parentCat property is not already defined for this node, it is probably
 	 * the root node. Non-root nodes get the property parentCat when this node's
 	 * children are marked below.
 	 */
+	options = options || {};
+
 	if (!mytree.hasOwnProperty('parentCat')){
 		mytree.parentCat = "is root"; //marks the root node
 	}
 
-	//mark maximal nodes
-	mytree.isMax = (mytree.cat !== mytree.parentCat);
+	//store the info of the most recent cat in order to skip over dummy nodes
+	//except if there is a dummy chain, then lastcat should be passed on the same
+	if(mytree.cat == "dummy"){
+		mytree.isMax = false;
+		mytree.isMin = false;
+		if(mytree.parentCat != "dummy"){
+			mytree.lastCat = mytree.parentCat;
+		}
+	}
+
+	//recall stored parentCat after dummies are skipped
+	if(mytree.parentCat == "dummy" && mytree.cat != "dummy"){
```

Comment:
I think this should be '===' instead of '==' and '!==' instead of '!=' (strict equality). Not sure it actually has an effect though.